test-data/unit/check-isinstance.test

```diff
@@ -1348,7 +1348,7 @@ l = [x for x in a if isinstance(x, int)]
 g = (x for x in a if isinstance(x, int))
 d = {0: x for x in a if isinstance(x, int)}
 reveal_type(l) # E: Revealed type is 'builtins.list[builtins.int*]'
-reveal_type(g) # E: Revealed type is 'typing.Iterator[builtins.int*]'
+reveal_type(g) # E: Revealed type is 'typing.Generator[builtins.int*]'
```

Comment:
Even though tests pass, I don't think this is right, because Generator takes three arguments, not one. I don't think there's a really elegant way to handle this, but somehow `check_generator_or_comprehension` needs to pass in all three args.

Reply:
That is a bit strange, changing the signatures to the proper `'typing.Generator[builtins.int*, builtins.None, builtins.None]'` makes the test fail, even though this type is used in other [places](https://github.com/python/mypy/blob/a2fd96280ba9127426f24c17bad351a16a5e1c70/test-data/unit/check-optional.test#L458).

Reply:
To be clear, you'll need to change checkexpr.py, not here—this just shows the wrong output.

Reply:
Yeah, I was just wondering why the correct signature does work in other places, but not here. The AsyncGenerator has even more examples where that is the case:

```
$ grep -r -n "typing.AsyncGenerator\[.*,.*\]" test-data/unit/
test-data/unit//check-async-await.test:218:    reveal_type(lst)  # E: Revealed type is 'typing.AsyncGenerator[builtins.int*, builtins.None]'
test-data/unit//check-async-await.test:220:    reveal_type(lst2)  # E: Revealed type is 'typing.AsyncGenerator[builtins.int*, builtins.None]'
test-data/unit//check-async-await.test:444:reveal_type(g)  # E: Revealed type is 'def () -> typing.AsyncGenerator[builtins.int, builtins.None]'
test-data/unit//check-async-await.test:445:reveal_type(g())  # E: Revealed type is 'typing.AsyncGenerator[builtins.int, builtins.None]'
test-data/unit//check-async-await.test:484:    reveal_type(gen.__aiter__())  # E: Revealed type is 'typing.AsyncGenerator[builtins.int*, builtins.None]'
```

Reply:
I guess mypy just doesn't check the number of arguments when you construct a Generator instance internally.